File: android/src/main/java/com/github/musicode/photobrowser/RNTPhotoBrowser.java
package com.github.musicode.photobrowser;

import android.view.Choreographer;
import android.view.View;

import com.facebook.react.uimanager.ThemedReactContext;
import com.github.herokotlin.photobrowser.PhotoBrowser;

class RNTPhotoBrowser extends PhotoBrowser {

    private Choreographer.FrameCallback frameCallback = new Choreographer.FrameCallback() {
        @Override
        public void doFrame(long frameTimeNanos) {
            for (int i = 0; i < getChildCount(); i++) {
                View child = getChildAt(i);
                child.measure(MeasureSpec.makeMeasureSpec(getMeasuredWidth(), MeasureSpec.EXACTLY),
                        MeasureSpec.makeMeasureSpec(getMeasuredHeight(), MeasureSpec.EXACTLY));
                child.layout(0, 0, child.getMeasuredWidth(), child.getMeasuredHeight());
            }
            getViewTreeObserver().dispatchOnGlobalLayout();
            Choreographer.getInstance().postFrameCallback(this);
        }
    };

    public RNTPhotoBrowser(ThemedReactContext reactContext) {
        super(reactContext);
        Choreographer.getInstance().postFrameCallback(frameCallback);
    }

    public void destroy() {
        Choreographer.getInstance().removeFrameCallback(frameCallback);
    }

}

File: android/src/main/java/com/github/musicode/photobrowser/RNTPhotoBrowser.kt
package com.github.musicode.photobrowser

import android.view.Choreographer

import com.facebook.react.uimanager.ThemedReactContext
import com.github.herokotlin.photobrowser.PhotoBrowser

class RNTPhotoBrowser(reactContext: ThemedReactContext) : PhotoBrowser(reactContext) {

    private val frameCallback = object : Choreographer.FrameCallback {
        override fun doFrame(frameTimeNanos: Long) {
            for (i in 0 until childCount) {
                val child = getChildAt(i)
                child.measure(MeasureSpec.makeMeasureSpec(measuredWidth, MeasureSpec.EXACTLY),
                        MeasureSpec.makeMeasureSpec(measuredHeight, MeasureSpec.EXACTLY))
                child.layout(0, 0, child.measuredWidth, child.measuredHeight)
            }
            viewTreeObserver.dispatchOnGlobalLayout()
            Choreographer.getInstance().postFrameCallback(this)
        }
    }

    init {
        Choreographer.getInstance().postFrameCallback(frameCallback)
    }

    fun destroy() {
        Choreographer.getInstance().removeFrameCallback(frameCallback)
    }

}